tsdb/docs/usage.md

```diff
@@ -0,0 +1,152 @@
+# Usage
+
+TSDB can be - and is - used by other applications such as [Cortex](https://cortexmetrics.io/) and [Thanos](https://thanos.io/).
+This directory contains documentation for any developers who wish to work on or with TSDB.
+
+Tip: `tsdb/db_test.go` demonstrates various usages of the TSDB library.
+
+## Instantiating a database
+
+Callers should use [`tsdb.Open`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#Open) to open a TSDB.
+(the directory may be new or pre-existing).
+This returns a [`*tsdb.DB`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB) which is the actual database.
+
+a `DB` has the following main components:
+
+* Compactor: a [leveled compactor](https://github.com/prometheus/prometheus/blob/c0c22ed04200a8d24d1d5719f605c85710f0d008/tsdb/compact.go#L78). Note: it is currently the only compactor implementation. It runs automatically.
+* [`Head`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB.Head)
+
+The `Head` is responsible for a lot.  Here are some of its main components:
+
+* [wal](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb/wal#WAL) (Write Ahead Log)
+* [`stripeSeries`](https://github.com/prometheus/prometheus/blob/1270b87970baeb926fcce64552db5c744ffaf83f/tsdb/head.go#L1279):
+  this holds all the active series by linking to [`memSeries`](https://github.com/prometheus/prometheus/blob/1270b87970baeb926fcce64552db5c744ffaf83f/tsdb/head.go#L1449)
+  by an ID (aka "ref") and by labels hash.
+* postings list (reverse index): For any label-value pair, holds all the corresponding series refs. Used for queries.
+* tombstones
+
+
+## Adding data
+
+Use [`db.Appender()`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB.Appender) to obtain an "appender".
+The [golang docs](https://pkg.go.dev/github.com/prometheus/prometheus/storage#Appender) speak mostly for themselves.
```

Comment:
uff, as pointed out previously, the golang docs have the wrong signature for the Appender. This could confuse developers more than it helps. Maybe we should fix the docs _before_ merging this PR (or make this and other godoc links point replaced with the codebase instead, pinned to the latest version and mention the version in this file).